packages/cli/src/executions/__tests__/execution-recovery.service.test.ts

```diff
@@ -170,7 +174,38 @@ describe('ExecutionRecoveryService', () => {
 				expect(amendedExecution).toBeNull();
 			});
 
-			test('should update `status`, `stoppedAt` and `data` if last node did not finish', async () => {
+			test('for successful dataless execution, should update `status`, `stoppedAt` and `data`', async () => {
+				/**
+				 * Arrange
+				 */
+				const workflow = await createWorkflow();
+				const execution = await createExecution(
+					{
+						status: 'success',
+						data: stringify(undefined), // saved execution but likely crashed while saving high-volume data
+					},
+					workflow,
+				);
+				const messages = setupMessages(execution.id, 'Some workflow');
+
+				/**
+				 * Act
+				 */
+				const amendedExecution = await executionRecoveryService.recoverFromLogs(
+					execution.id,
+					messages,
+				);
+
+				/**
+				 * Assert
+				 */
+				assert(amendedExecution);
+				expect(amendedExecution.stoppedAt).not.toBe(execution.stoppedAt);
```

Comment:
This is an edge case where the execution did stop (executing) but did not finish (saving). The execution recovery service updates `stoppedAt` based on logs - even though in this edge case strictly speaking there is no need to update `stoppedAt` based on logs, it is simpler to allow the update, rather than prevent it and make this service even more complex.